syntax = "proto3";
package Protocol;

import "Enum.proto";
import "Struct.proto";

message CTS_LOGIN
{

}

message STC_LOGIN
{
	bool success = 1;
	repeated ObjectInfo players = 2;
}

message CTS_ENTER_GAME
{
	uint64 playerIndex = 1;
}

message STC_ENTER_GAME
{
	bool success = 1;
	ObjectInfo player = 2;
}

message CTS_LEAVE_GAME
{

}

message STC_LEAVE_GAME
{

}

message STC_SPAWN
{
	repeated ObjectInfo players = 1;
}

message STC_DESPAWN
{
	repeated uint64 object_ids = 1;
}

message CTS_MOVE
{
	PosInfo info = 1;
}

message STC_MOVE
{
	PosInfo info = 1;
}

message CTS_CHAT
{
	string msg = 1;
}

message STC_CHAT
{
	uint64 playerId = 1;
	string msg = 2;
}